### DISCUSSION TITLE
How to use `tracing` in across dylibs

### DISCUSSION THREAD

**Discussion by JakkuSakura** [tokio-rs/tracing#1479] [Q&A] [answered]
https://github.com/rust-lang/log/issues/421#issuecomment-885764004
I have a solution for `log` facade, with `tracing-log`, but what if I want to use full functionality of `tracing`? This approach will work if I figure out how to forward full functionality of `tracing`(mainly macros that involves global variables)

**Comment by davidbarsky** [comment]
I'm assuming that the usage of tracing across dylibs implies exposing tracing-specific functionality through a C FFI. In that case, I suggest looking at https://github.com/tokio-rs/tracing/issues/948 for some prior discussion on the subject.

**Comment by JakkuSakura** [comment, marked as answer]
https://github.com/rust-lang/log/issues/421#issuecomment-885764004

**Comment by JakkuSakura** [comment]
A crate for this purpose is available. No unsafe is used at all. Also support logs
https://crates.io/crates/tracing-shared
